### PR TITLE
OWLS-84517: Scaling failed when setting Dedicated to true

### PR DESCRIPTION
**Pull request by lennyphan**
When namespace selection strategy is configured to be 'Dedicated', requests to the Operator's REST endpoint fail authentication.  Authentication checks were always cluster scoped.  The fix is for REST endpoint authentication to be namespace scoped when in 'Dedicated' namespace selection strategy,

### CODE REVIEW

File: operator/src/test/java/oracle/kubernetes/operator/helpers/AuthenticationProxyTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    public boolean check(
+        String principal,
+        Operation operation,
+        Resource resource,
+        String resourceName,
+        Scope scope,
+        String namespaceName) {
+      this.scope = scope;
+      System.out.println("AuthorizationProxyStub.check scope: " + scope);
```
Comment (russgold) [inline operator/src/test/java/oracle/kubernetes/operator/helpers/AuthenticationProxyTest.java:68]: It is bad practice to leave a println in your tests. I assume you needed this for debugging and forgot to remove it?

**Review by russgold — CHANGES_REQUESTED** [review]
Other than the one oversight, this looks good.

**Comment by hzhao-github** [comment]
With the change, Scaling test passed in my local:

<09-15-2020 13:06:30> <INFO> <oracle.weblogic.kubernetes.extensions.IntegrationTestWatcher printHeader> <
+++++++++++++++++++++++++++++++++++++++++++++++++
Test PASSED testDedicatedModeSameNamespaceScale()
+++++++++++++++++++++++++++++++++++++++++++++++++
>
<09-15-2020 13:06:30> <INFO> <oracle.weblogic.kubernetes.extensions.IntegrationTestWatcher printHeader> <
----------------------------------------------------------------
Starting afterAll for oracle.weblogic.kubernetes.ItDedicatedMode
----------------------------------------------------------------

**Review by alai8 — APPROVED** [review]

**Review by hzhao-github — APPROVED** [review]

**Comment by hzhao-github** [comment]
LGTM

**Review by russgold — APPROVED** [review]

**Pull request merged**